Vector/AppDelegate.m

```diff
@@ -739,6 +722,225 @@ - (void)refreshApplicationIconBadgeNumber
     [UIApplication sharedApplication].applicationIconBadgeNumber = [MXKRoomDataSourceManager notificationCount];
 }
 
+#pragma mark - Universal link
+
+- (BOOL)handleUniversalLink:(NSUserActivity*)userActivity
+{
+    NSURL *webURL = userActivity.webpageURL;
+    NSLog(@"[AppDelegate] handleUniversalLink: %@", webURL.absoluteString);
+
+    return [self handleUniversalLinkFragment:webURL.fragment];
+}
+
+- (BOOL)handleUniversalLinkFragment:(NSString*)fragment
+{
+    BOOL continueUserActivity = NO;
+    MXKAccountManager *accountManager = [MXKAccountManager sharedManager];
+
+    NSLog(@"[AppDelegate] Universal link: handleUniversalLinkFragment: %@", fragment);
+
+    // The app manages only one universal link at a time
+    // Discard any pending one
+    [self resetPendingUniversalLink];
+
+    // Extract params
+    NSArray<NSString*> *pathParams;
+    NSMutableDictionary *queryParams;
+    [self parseUniversalLinkFragment:fragment outPathParams:&pathParams outQueryParams:&queryParams];
+
+    // Check the action to do
+    if ([pathParams[0] isEqualToString:@"room"] && pathParams.count >= 2)
+    {
+        if (accountManager.activeAccounts.count)
+        {
+            // The link is the form of "/room/[roomIdOrAlias]" or "/room/[roomIdOrAlias]/[eventId]"
+            NSString *roomIdOrAlias = pathParams[1];
+
+            // Is it a link to an event of a room?
+            NSString *eventId = (pathParams.count >= 3) ? pathParams[2] : nil;
+
+            // Check there is an account that knows this room
+            MXKAccount *account = [accountManager accountKnowingRoomWithRoomIdOrAlias:roomIdOrAlias];
```

Comment:
A link to a public room, or even a private room, will open the room preview asking if the user wants to join it.
This will be done in the "email invitation link" coming PR.